src/csharp/Microsoft.Spark/ML/Feature/SQLTransformer.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.ML.Feature
+{
+    /// <summary>
+    /// A <see cref="SQLTransformer"/> implements the transformations which are defined by SQL statement.
+    /// </summary>
+    public class SQLTransformer : FeatureBase<SQLTransformer>, IJvmObjectReferenceProvider
+    {
+        private static readonly string s_sqlTransformerClassName =
+    "org.apache.spark.ml.feature.SQLTransformer";
+
+        /// <summary>
+        /// Create a <see cref="SQLTransformer"/> without any parameters.
+        /// </summary>
+        public SQLTransformer() : base(s_sqlTransformerClassName)
+        {
+        }
+
+        /// <summary>
+        /// Create a <see cref="SQLTransformer"/> with a UID that is used to give the
+        /// <see cref="SQLTransformer"/> a unique ID.
+        /// </summary>
+        /// <param name="uid">An immutable unique ID for the object and its derivatives.</param>
+        public SQLTransformer(string uid) : base(s_sqlTransformerClassName, uid)
+        {
+        }
+
+        internal SQLTransformer(JvmObjectReference jvmObject) : base(jvmObject)
+        {
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Executes the <see cref="SQLTransformer"/> and transforms the DataFrame to include the new
+        /// column.
+        /// </summary>
+        /// <param name="source">The DataFrame to transform</param>
+        /// <returns>
+        /// New <see cref="DataFrame"/> object with the source <see cref="DataFrame"/> transformed
+        /// </returns>
+        public DataFrame Transform(DataFrame source) =>
```

Comment:
For consistency, is it possible to add TransformSchema as well as Transform? https://spark.apache.org/docs/2.1.0/api/java/org/apache/spark/ml/feature/SQLTransformer.html#transformSchema(org.apache.spark.sql.types.StructType)